membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -13,19 +13,31 @@ import models._
 import monitoring.Metrics
 import play.api.libs.concurrent.Execution.Implicits._
 import play.api.libs.json.Json
-import play.api.mvc.{Controller, Result}
+import play.api.mvc._
 import play.filters.cors.CORSActionBuilder
 
-import scala.concurrent.Future
+import scala.concurrent.{Future, Promise}
 import scalaz.std.scalaFuture._
 import scalaz.syntax.std.option._
 import scalaz.{EitherT, \/}
+import configuration.Config.authentication
 
 
 class AttributeController extends Controller with LazyLogging {
 
+  def signed(): ActionBuilder[Request] = new ActionBuilder[Request] {
+    def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]) = {
+      if (request.headers.get("authentication").contains(authentication.key)) {
+        block(request)
+      } else {
+        Future.successful(Forbidden)
+      }
+    }
+  }
+
   lazy val corsFilter = CORSActionBuilder(Config.corsConfig)
   lazy val backendAction = NoCacheAction andThen corsFilter andThen BackendFromCookieAction
+  lazy val backendForSyncWithZuora = NoCacheAction andThen  signed andThen WithBackendFromUserIdAction
```

Comment:
After talking with @adamnfish, he mentioned that as an advantage of having the signed request we can reduce the surface of a DOS attack because we won't hit Zuora or Salesforce if the request is not valid. At the same time, this introduces an extra layer of maintenance because if the shared secret changes a lot, we should update all the app that use this endpoint.